components/Login/RegisterForm.tsx

```diff
@@ -1,59 +1,57 @@
 import { useEffect, useState } from "react";
-import { View } from "react-native";
-import { loginUser, registerUser } from "../../api/user-api";
-import FormTextField from "../General/Forms/FormTextField";
-import FormButton from "../General/Forms/FormButton";
-import PasswordEye from "../General/Forms/PasswordEye";
+import { View, Text } from "react-native";
+import { loginUser, registerUser } from "@/api/user-api";
+import FormTextField from "@/components/General/Forms/FormTextField";
+import FormButton from "@/components/General/Forms/FormButton";
 import { MaterialCommunityIcons } from "@expo/vector-icons";
-import ShowAlert from "../General/ShowAlert";
-import FormFieldAlert from "../General/Forms/FormFieldAlert";
+import ShowAlert from "@/components/General/ShowAlert";
+import FormFieldAlert from "@/components/General/Forms/FormFieldAlert";
 import {
   removeEmojis,
   validatePasswordContainsLetter,
   validatePasswordLength,
   validateEmail,
   validateName,
-} from "../General/validation";
-import Text from "../General/Text";
-import errorSwitch from "../General/error-switch";
+} from "@/components/General/validation";
+import errorSwitch from "@/components/General/error-switch";
 import { useNavigation } from "@react-navigation/native";
-import DialogNotification from "../General/DialogNotification";
+import DialogNotification from "@/components/General/DialogNotification";
 import { AlertNotificationRoot } from "react-native-alert-notification";
-import tailwindConfig from "@/tailwind.config";
-import { setUserInfo, setJWT } from "../../services/storage-service";
+import { colors } from "@/theme/colors";
+import { setUserInfo, setJWT } from "@/services/storage-service";
+import { UserInfo } from "@/types/user";
+import { isAxiosError } from "axios";
+
+interface BaseUserType {
+  _id: string;
+  user: UserInfo;
+}
 
 /**
  * Component for registering a new account in the system, used in the register screen
- * @returns {React.Element} Component containing the form for registering a new user
  */
-
-export default function RegisterForm() {
-  const tailwindColors = tailwindConfig.theme.colors;
-
+export const RegisterForm = () => {
   const navigation = useNavigation();
-  const [firstName, setFirstName] = useState("");
-  const [lastName, setLastName] = useState("");
-  const [email, setEmail] = useState("");
-  const [password, setPassword] = useState("");
-  const [confirmPassword, setConfirmPassword] = useState("");
 
-  const [emailAlert, setEmailAlert] = useState("");
-  const [nameAlert, setNameAlert] = useState("");
-  const [isAllInputValid, setIsAllInputValid] = useState(false);
-  const [confirmPasswordAlert, setConfirmPasswordAlert] = useState("");
+  const [name, setName] = useState<string>("");
+  const [email, setEmail] = useState<string>("");
+  const [password, setPassword] = useState<string>("");
+  const [confirmPassword, setConfirmPassword] = useState<string>("");
```

Comment:
No need for explicit typing as Typescript can infer.
```suggestion
  const [name, setName] = useState("");
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const [confirmPassword, setConfirmPassword] = useState("");
```